src/main/java/edu/illinois/cs/dt/tools/fixer/CleanerFixerPlugin.java

```diff
@@ -186,13 +183,97 @@ private void applyFix(final String polluterName,
+    // In addition to polluter and victim to run, need also list of statements; assumed statements already make victim pass
+    // The value n is needed to represent granularity of partitioning the statements to get minimal list
+    private NodeList<Statement> deltaDebug(final String polluterName,
+                                           final JavaMethod victimMethod,
+                                           final NodeList<Statement> cleanerStmts,
+                                           int n) throws Exception {
+        // If n granularity is greater than number of cleaner statements, then finished, simply return cleaner statements
+        if (cleanerStmts.size() < n) {
+            return cleanerStmts;
+        }
+
+        // Cut the statements into n equal chunks and try each chunk
+        int chunkSize = (int)Math.round((double)(cleanerStmts.size()) / n);
+        List<NodeList<Statement>> chunks = new ArrayList<>();
+        for (int i = 0; i < cleanerStmts.size(); i += chunkSize) {
+            NodeList<Statement> chunk = NodeList.nodeList();
+            NodeList<Statement> otherChunk = NodeList.nodeList();
+            // Start complement chunk by grabbing statements before chunk start
+            for (int j = 0; j < i; j++) {
+                otherChunk.add(cleanerStmts.get(j));
+            }
+            // Create chunk starting at this iteration
+            for (int j = i; j < Math.min(cleanerStmts.size(), i + chunkSize); j++) {
+                chunk.add(cleanerStmts.get(j));
+            }
+            // Complete complement chunk by grabbing statements after
+            for (int j = Math.min(cleanerStmts.size(), i + chunkSize); j < cleanerStmts.size(); j++) {
+                otherChunk.add(cleanerStmts.get(j));
+            }
+
+            // Check if applying chunk works
+            if (checkCleanerStmts(polluterName, victimMethod, chunk)) {
+                return deltaDebug(polluterName, victimMethod, chunk, 2);    // If works, then delta debug some more this chunk
+            }
+            // Otherwise, check if applying complement chunk works
+            if (checkCleanerStmts(polluterName, victimMethod, otherChunk)) {
+                return deltaDebug(polluterName, victimMethod, otherChunk, n - 1);   // If works, then delta debug some more the complement chunk
+            }
+        }
+        // If not chunk/complement work, increase granularity and try again
+        return deltaDebug(polluterName, victimMethod, cleanerStmts, n * 2);
+    }
+
+    // TODO: Extract this logic out to a more generalized fixer that is separate, so we can reuse it
+    // TODO: for cleaners, santa clauses, etc.
+    private void applyFix(final String polluterName,
+                          final JavaMethod cleanerMethod,
+                          final JavaMethod victimMethod) throws Exception {
+        backup(victimMethod.javaFile());
+
+        // Compile the test without our fix
+        runMvnInstall();
```

Comment:
I believe I originally added this line in a previous commit, but it seems to me this is actually a waste of time because we're running inside a Maven plugin which already uses the `test-compile` goal, so we already know that everything works with no changes.